src/solvers/smt2/smt2_solver.cpp

```diff
@@ -81,39 +81,42 @@ void smt2_solvert::expand_function_applications(exprt &expr)
   {
     auto &app=to_function_application_expr(expr);
 
-    // look it up
-    irep_idt identifier=app.function().get_identifier();
-    auto f_it=id_map.find(identifier);
-
-    if(f_it!=id_map.end())
+    if(app.function().id() == ID_symbol)
```

Comment:
So nothing happens if it's not a symbol?

Reply:
Yes, only function symbols need to be expanded.